crates/exec-wasmtime/src/lib.rs

```diff
@@ -43,17 +42,53 @@ pub enum Package {
     },
 }
 
+/// Package to execute
+#[cfg(windows)]
+#[derive(Debug)]
+pub enum Package {
+    /// Remote URL to fetch package from
+    Remote(Url),
+
+    /// Local package
+    Local {
+        /// Open WASM module file Handle
+        wasm: std::fs::File,
+        /// Optional open config file descriptor
+        conf: Option<std::fs::File>,
+    },
+}
+
 /// The Arguments
 // NOTE: `repr(C)` is required, otherwise `toml` serialization fails with `values must be emitted before tables`
-#[derive(Debug, Deserialize, Serialize)]
+#[derive(Debug)]
+#[cfg_attr(unix, derive(Deserialize, Serialize))]
 #[repr(C)]
 pub struct Args {
     /// Package
     pub package: Package,
 }
 
+/// Execute
+pub fn execute_with_args(args: Args) -> anyhow::Result<()> {
+    // Step through the state machine.
+    let configured = Loader::from(args);
+    let requested = configured.next()?;
+    let attested = requested.next()?;
+    let compiled = attested.next()?;
+    let connected = compiled.next()?;
+    let completed = connected.next()?;
+    drop(completed);
+    Ok(())
+}
+
+#[cfg(unix)]
 /// Execute
```

Comment:
Let's document that this assumes a Unix socket on fd 3 and reads and parses args from there.

Reply:
done